### PR TITLE
BUG: runtime error caused by read_csv and read_parquet with relative path

### LINKED ISSUE #271: BUG: runtime error caused by `read_xxx` with relative path
### Describe the bug
In the interactive environment, the work dir of a local cluster is determined during initialization. If the current dir of IPython changed, loading a dataframe with a relative path may result in a FileNotFound error.

### To Reproduce
```python
In [1]: import xorbits

In [2]: xorbits.init()
2023-03-14 21:20:14,070 xorbits._mars.deploy.oscar.local 22259 WARNING  Web service started at http://0.0.0.0:45351

In [3]: cd /path/to/my/dataset

In [4]: import xorbits.pandas as pd

In [5]: trips = pd.read_parquet("data.parquet")  # will cause a FileNotFoundError since Xorbits is initalized under another directory

In [6]: trips 
```

### Expected behavior
Xorbits should better take the relative path and make it an absolute path.

### PR DESCRIPTION
**Pull request by aa452948257**
## What do these changes do?
 Fix runtime error caused by `read_csv` with relative path.

## Related issue number
Resolves #271 

## Check code requirements

- [x] tests added / passed (if needed)
- [ ] Ensure all linting tests pass

### CODE REVIEW

File: python/xorbits/_mars/dataframe/datasource/tests/test_datasource.py [hunk 1]
```diff
@@ -485,6 +487,35 @@ def test_read_csv():
...
+    try:
+        df2 = df1 = pd.DataFrame(
+            np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]]),
+            columns=["a", "b", "c"],
+            dtype=np.int64,
+        )
+        df1.to_csv(file_path_1)
+        df2.to_csv(file_path_2)
+        mdf = read_csv(tempdir, index_col=0, chunk_bytes=10)
+        print(mdf)
```
Comment (qianduoduo0904) [inline python/xorbits/_mars/dataframe/datasource/tests/test_datasource.py:503]: `print` should be removed.

File: python/xorbits/_mars/dataframe/datasource/read_csv.py [hunk 2]
```diff
@@ -739,8 +739,25 @@ def read_csv(
     columns_value = parse_index(mini_df.columns, store_data=True)
     if index_col and not isinstance(index_col, int):
         index_col = list(mini_df.columns).index(index_col)
+
+    # convert path to abs_path
+    if isinstance(path, (list, tuple)):
```
Comment (qianduoduo0904) [inline python/xorbits/_mars/dataframe/datasource/read_csv.py:744]: Did you have test for this situation?

Comment (aa452948257) [reply python/xorbits/_mars/dataframe/datasource/read_csv.py:744]: Yes, i have tested this situation.

File: python/xorbits/_mars/dataframe/datasource/read_csv.py [hunk 3]
```diff
@@ -686,6 +686,7 @@ def read_csv(
     >>> # read from OSS
     >>> auth_path = build_oss_path(file_path, access_key_id, access_key_secret, end_point)
     >>> md.read_csv(auth_path)
+    s3://buvket/file.txt
```
Comment (UranusSeven) [inline python/xorbits/_mars/dataframe/datasource/read_csv.py:689]:
```suggestion
    s3://bucket/file.txt
```

File: python/xorbits/_mars/dataframe/datasource/read_csv.py (hunk 2 above)
Comment (qianduoduo0904) [reply python/xorbits/_mars/dataframe/datasource/read_csv.py:744]: I mean the ut, which UT covered this branch?

File: python/xorbits/_mars/dataframe/datasource/utils.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# Copyright 2022-2023 XProbe Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (UranusSeven) [inline python/xorbits/_mars/dataframe/datasource/utils.py:2]:
```suggestion

# Licensed under the Apache License, Version 2.0 (the "License");
```

File: python/xorbits/_mars/dataframe/datasource/utils.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from ...lib.filesystem import LocalFileSystem, get_fs, glob
+
+
+def convert_to_abspath(path, storage_options):
```
Comment (UranusSeven) [inline python/xorbits/_mars/dataframe/datasource/utils.py:19]: Better add type hints. Plz refer: https://docs.python.org/zh-cn/3/library/typing.html

File: python/xorbits/_mars/dataframe/datasource/read_csv.py (hunk 2 above)
Comment (aa452948257) [reply python/xorbits/_mars/dataframe/datasource/read_csv.py]: I added some UTs in current commit.

File: python/xorbits/_mars/dataframe/datasource/read_csv.py [hunk 6]
```diff
@@ -686,6 +686,7 @@ def read_csv(
     >>> # read from OSS
     >>> auth_path = build_oss_path(file_path, access_key_id, access_key_secret, end_point)
     >>> md.read_csv(auth_path)
+    s3://bucket/file.txt
```
Comment (UranusSeven) [inline python/xorbits/_mars/dataframe/datasource/read_csv.py:689]:
```suggestion
```

File: python/xorbits/_mars/dataframe/datasource/tests/test_datasource.py [hunk 7]
```diff
@@ -469,6 +469,8 @@ def test_read_csv():
         )
         df.to_csv(file_path)
         mdf = read_csv(file_path, index_col=0, chunk_bytes=10)
+        # mdf.op.path
```
Comment (UranusSeven) [inline python/xorbits/_mars/dataframe/datasource/tests/test_datasource.py:472]:
```suggestion
```

File: python/xorbits/_mars/dataframe/datasource/tests/test_datasource.py [hunk 8]
```diff
@@ -485,6 +487,62 @@ def test_read_csv():
...
+        df2 = df1 = pd.DataFrame(
+            np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]]),
+            columns=["a", "b", "c"],
+            dtype=np.int64,
+        )
+        df1.to_csv(file_path_1)
+        df2.to_csv(file_path_2)
+        mdf = read_csv(tempdir, index_col=0, chunk_bytes=10)
+        assert isinstance(mdf.op, DataFrameReadCSV)
+        assert mdf.shape[1] == 3
```
Comment (UranusSeven) [inline python/xorbits/_mars/dataframe/datasource/tests/test_datasource.py:504]: Need to assert if mdf.op.path is abs or not.

File: python/xorbits/pandas/pandas_adapters/core.py [hunk 9]
```diff
@@ -32,7 +32,7 @@
 from ...core.utils.fallback import wrap_fallback_module_method
 
 _NO_ANNOTATION_FUNCS: Dict[Callable, MarsOutputType] = {
-    pd.read_pickle: MarsOutputType.dataframe,
+    pd.read_pickle: MarsOutputType.object,
```
Comment (UranusSeven) [inline python/xorbits/pandas/pandas_adapters/core.py:35]: Better fix this issue in another PR.

File: python/xorbits/_mars/dataframe/datasource/utils.py (hunk 5 above)
Comment (aa452948257) [reply python/xorbits/_mars/dataframe/datasource/utils.py]: Fixed

File: python/xorbits/_mars/dataframe/datasource/tests/test_datasource.py [hunk 10]
```diff
@@ -485,6 +487,62 @@ def test_read_csv():
...
+    try:
+        df2 = df1 = pd.DataFrame(
+            np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]]),
+            columns=["a", "b", "c"],
+            dtype=np.int64,
+        )
+        df1.to_csv(file_path_1)
+        df2.to_csv(file_path_2)
+        mdf = read_csv([file_path_1, file_path_2], index_col=0, chunk_bytes=10)
+        assert isinstance(mdf.op, DataFrameReadCSV)
```
[the remark this answers is not in the document]
Comment (aa452948257) [reply python/xorbits/_mars/dataframe/datasource/tests/test_datasource.py:530]: Added

File: python/xorbits/_mars/dataframe/datasource/read_csv.py [hunk 11]
```diff
@@ -686,6 +686,8 @@ def read_csv(
     >>> # read from OSS
     >>> auth_path = build_oss_path(file_path, access_key_id, access_key_secret, end_point)
     >>> md.read_csv(auth_path)
+>>> # read from S3
+>>> md.read_csv('s3://bucket/file.txt')
```
Comment (UranusSeven) [inline python/xorbits/_mars/dataframe/datasource/read_csv.py:690]:
```suggestion
    >>> # read from S3
    >>> md.read_csv('s3://bucket/file.txt')
```

**Review by aa452948257 — COMMENTED** [review]
Fixed

**Review by UranusSeven — APPROVED** [review]
LGTM

File: python/xorbits/_mars/dataframe/datasource/utils.py [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from ...lib.filesystem import LocalFileSystem, get_fs
+
+
+def convert_to_abspath(path: str, storage_options: dict = None):
```
Comment (ChengjieLi28) [inline python/xorbits/_mars/dataframe/datasource/utils.py:20]: According to your implementation, the type of parameter "path" can be list, tuple, or string. Therefore, please provide the complete type annotation.

Comment (ChengjieLi28) [reply python/xorbits/_mars/dataframe/datasource/utils.py:20]: Also, the type annotation for parameter "storage_options" should be "dict | None"

File: python/xorbits/_mars/dataframe/datasource/tests/test_datasource.py [hunk 13]
```diff
@@ -485,6 +486,67 @@ def test_read_csv():
...
+        for chunk in mdf.chunks:
+            index_keys.add(chunk.index_value.key)
+            pd.testing.assert_index_equal(df1.columns, chunk.columns_value.to_pandas())
+            pd.testing.assert_series_equal(df1.dtypes, chunk.dtypes)
+        assert len(index_keys) > 1
+    finally:
+        shutil.rmtree(tempdir)
+
+
+def test_read_csv_list():
```
Comment (ChengjieLi28) [inline python/xorbits/_mars/dataframe/datasource/tests/test_datasource.py:519]: nit: Should we add UT for testing the situation of reading csv and parquet file via GPU when using relative path?

File: python/xorbits/_mars/dataframe/datasource/utils.py (hunk 12 above)
Comment (aa452948257) [reply python/xorbits/_mars/dataframe/datasource/utils.py:20]: Ok

**Review by ChengjieLi28 — APPROVED** [review]

**Review by UranusSeven — APPROVED** [review]

**Pull request merged**